src/main/java/spoon/reflect/visitor/chain/CtQuery.java

```diff
@@ -23,59 +23,91 @@
 
 /**
  * CtQuery represents a query, which can be used to traverse a spoon model and collect
- * children elements in several ways.<br>
+ * children elements in several ways.<br/>
  *
- * <br>
- * Use {@link CtQueryable#map(CtFunction)}} or {@link CtQueryable#filterChildren(Filter)} to create a new query starting from an existing element.<br>
+ * Creation: A query is created either from a {@link CtElement}, or it can be used defined first from {@link CtQueryImpl} and bound to root elements
+ * afterwards using {{@link #setInput(Object...)}.<br/>
  *
- * The main methods are:
- * <ul>
- * <li> {@link #map(CtFunction))} - uses a lambda expression to return any model elements that are directly accessible from an input element.
- * <li> {@link #map(CtConsumableFunction))} -implementations of {@link CtConsumableFunction} provides a complex queries.
- * <li> {@link #filterChildren(Filter))} - uses {@link Filter} instances to filter children of an element
- * <li> {@link #list()} - to evaluate the query and return a list of elements produced by this query.
- * </ul>
- * It makes sense to evaluate this query only once, because the input element is constant.<br>
- * A CtQuery is lazily evaluated once {{@link #list()}} or {@link #forEach(CtConsumer)} are called.
- * Usually a new query is created each time when one needs to query something.
- * If you need to reuse a query instance several times, for example in a loop, then use {@link CtQuery#setInput(Object...)}
- * to bound this query with different input.
+ * Chaining: In a query several steps can be chained, by chaining calls to map functions. The non-null outputs of one step
+ * are given as input to the next step. An iterable or array output is considered as a set of different inputs for the next step.
+ *
+ * Evaluation: A CtQuery is lazily evaluated once {{@link #list()}} or {@link #forEach(CtConsumer)} are called.<br/>
  *
  */
 public interface CtQuery extends CtQueryable {
 
 	/**
-	 * sets (binds) the input of the query. If the query is created by {@link CtElement#map} or {@link CtElement#filterChildren(Filter)},
+	 * Recursively scans all children elements of an input element.
+	 * The matched child element for which (filter.matches(element)==true) are sent to the next query step.
+	 * Essentially the same as {@link CtElement#getElements(Filter)} but more powerful, because it
+	 * can be chained with other subsequent queries.
+	 *
+	 * Note: the input element (the root of the query, `this` if you're in {@link CtElement}) is also checked and may thus be also sent to the next step.
+	 * The elements which throw {@link ClassCastException} during {@link Filter#matches(CtElement)}
+	 * are considered as **not matching**, ie. are excluded.
+	 *
+	 * @param filter used to filter scanned children elements of the AST tree
+	 * @return a new Query

```

Comment:
just note... do not know if relevant ...
it is "a new Query" only when called first time on CtElement. Later the same query instance is returned (this)